src/main/java/org/zaproxy/zap/extension/jwt/JWTExtension.java

```diff
@@ -31,25 +48,77 @@
 public class JWTExtension extends ExtensionAdaptor {
 
     protected static final Logger LOGGER = Logger.getLogger(JWTExtension.class);
+    private static final List<Class<? extends Extension>> DEPENDENCIES;
+
+    static {
+        List<Class<? extends Extension>> dependencies = new ArrayList<>(1);
+        dependencies.add(ExtensionHttpFuzzer.class);
+        DEPENDENCIES = Collections.unmodifiableList(dependencies);
+    }
+
+    private HttpFuzzerHandler httpFuzzerHandler;
+    private JWTMessageLocationReplacerFactory jwtMessageLocationReplacerFactory;
+
+    @Override
+    public List<Class<? extends Extension>> getDependencies() {
+        return DEPENDENCIES;
+    }
 
     @Override
     public String getAuthor() {
         return "KSASAN preetkaran20@gmail.com";
     }
 
+    @Override
+    public void init() {
+        JWTI18n.init();
+        jwtMessageLocationReplacerFactory = new JWTMessageLocationReplacerFactory();
+        MessageLocationReplacers.getInstance()
+                .addReplacer(HttpMessage.class, jwtMessageLocationReplacerFactory);
+    }
+
+    @Override
+    public void initView(ViewDelegate view) {
+        super.initView(view);
+
+        HttpPanelManager panelManager = HttpPanelManager.getInstance();
+        panelManager.addRequestViewFactory(
+                RequestAllComponent.NAME, new JWTFuzzPanelViewFactory(null));
+        panelManager.addRequestViewFactory(
+                RequestSplitComponent.NAME, new JWTFuzzPanelViewFactory(ViewComponent.HEADER));
+        panelManager.addRequestViewFactory(
+                RequestSplitComponent.NAME, new JWTFuzzPanelViewFactory(ViewComponent.BODY));
+    }
+
     @Override
     public void hook(ExtensionHook extensionHook) {
         super.hook(extensionHook);
-        JWTI18n.init();
         try {
             extensionHook.addOptionsParamSet(getJWTConfiguration());
             extensionHook.getHookView().addOptionPanel(new JWTOptionsPanel());
+            ExtensionFuzz extensionFuzz =
+                    Control.getSingleton().getExtensionLoader().getExtension(ExtensionFuzz.class);
+            extensionFuzz.addFuzzerHandler(httpFuzzerHandler);
             LOGGER.debug("JWT Extension loaded successfully");
         } catch (Exception e) {
             LOGGER.error("JWT Extension can't be loaded. Configuration not found or invalid", e);
         }
     }
 
+    @Override
+    public void unload() {
+        super.unload();
+        HttpPanelManager panelManager = HttpPanelManager.getInstance();
```

Comment:
For correctness the views need to be removed as well.

Reply:
the views created by the factory ?

Reply:
Yes.